atef/widgets/config.py

```diff
@@ -186,7 +186,7 @@ def save_as(self, *args, filename: Optional[str] = None, **kwargs):
                 filter='Json Files (*.json)',
             )
         try:
-            with open(filename, 'w') as fd:
+            with open(filename+'.json', 'w') as fd:
```

Comment:
What happens if I type in `my_filename.json` for the filename? Will it become `my_filename.json.json`?

Reply:
Yes it will. But the goal of this bug fix was so that atef would add the extension automatically in order to be consistent with the "Files of Type" so the user wouldn't need to. atef will still be able to open your file as a .json file.

Reply:
I was hoping to prompt you into realizing that the behavior I described is undesirable, but let me put it more succinctly:

* If `.json` is already the suffix of the filename, `.json` should not be added. Multiple filename extensions that have no meaning are at best undesirable or confusing.
* If `.json` is not the suffix of the filename, it should be concatenated to the end of the filename string

If you try a 'save' dialog in something like MS Paint (Word, etc) you'll find that it performs as I described above.

Consider looking at `pathlib`:

```python
In [1]: import pathlib

In [2]: filename = pathlib.Path("/path/to/temp")

In [3]: filename
Out[3]: PosixPath('/path/to/temp')

In [4]: filename.suffix
Out[4]: ''

In [5]: filename = pathlib.Path("/path/to/temp.json")

In [6]: filename.suffix
Out[6]: '.json'
```

Reply:
I changed it so that I added a check to see if `filename` ends in `.json` instead of always adding the `.json` extension.